client/components/select-dropdown/style.scss

```diff
@@ -54,11 +54,30 @@
 			right: 13px;
 			top: 12px;
 
+		.is-compact & {
+			right: 4px;
+			top: 4px;
+		}
+
 		display: block;
 		line-height: 18px;
 		color: rgba( $gray, 0.5 );
 	}
 
+
+	.is-compact & {
+		padding: 7px;
+		color: darken( $gray, 10% );
+		font-size: 11px;
+		line-height: 1;
+		text-transform: uppercase;
+
+		.count {
```

Comment:
Agreed, let's ditch the border of count for the compact dropdown only. I think it's ok to leave the regular sized one the same.